### ISSUE TITLE
t-test giving wrong output for 95% CI

### ISSUE THREAD

**Issue by reneshbedre** [raphaelvallat/pingouin#119]
Hi,

Thank you for the great package. I was doing one-sample t-test using the `pingouin` as follows,

`from pingouin import ttest`
`ttest([5.5, 2.4, 6.8, 9.6, 4.2], 4).round(2)`

Output,
          T  dof       tail  p-val          CI95%  cohen-d   BF10  power
T-test  1.4    4  two-sided   0.23  **[-1.68, 5.08]**     0.62  0.766   0.19

I did the same in R and the output for 95% CI does not match,

`t.test(c(5.5, 2.4, 6.8, 9.6, 4.2), mu=4)`

Output,

	One Sample t-test

data:  c(5.5, 2.4, 6.8, 9.6, 4.2)
t = 1.3974, df = 4, p-value = 0.2348
alternative hypothesis: true mean is not equal to 4
95 percent confidence interval:
 **2.322309 9.077691**
sample estimates:
mean of x 
      5.7 

Is there something I am missing here?

**Comment by raphaelvallat** [comment]
Hi @reneshbedre,

Thanks so much for opening the issue and checking your results against R, Pingouin is still a young library and there's still a lot of work to validate all functions/scenarios. 

You were absolutely right, there was an error in how the confidence intervals were calculated for one-sample T-test with a true mean different than zero (e.g. `y = 4`). Let me emphasize that this error did not affect the confidence intervals of any other T-tests: the one-sample against a true mean of zero, paired two-samples T-test, independent T-tests (with or without Welch correction) all gave valid confidence intervals. 

Based on the R code of the t.test function, the solution is simply to add the true mean to the confidence interval, e.g.:

```
conf = 0.975 if tail == 'two-sided' else 0.95
tcrit = t.ppf(conf, dof)
ci = np.array([tval - tcrit, tval + tcrit]) * se
if ny == 1:
    ci += y
```

I have now fixed the issue on the latest commit of the develop branch, and I've also updated and improved the unit testing of the pingouin.ttest function against the R t.test function, as you can see here: https://github.com/raphaelvallat/pingouin/commit/fa05c8394beef6ca158be21cc2b8eee18796694e

I will release a new hotfix version in the upcoming days.

Thanks again,
Raphael

**Comment by reneshbedre** [comment]
Thank you for the update and fixing the issue.

**Comment by raphaelvallat** [comment]
This is fixed in the just-released stable version of Pingouin (v0.3.8). Please make sure to update your version. Closing the issue. Thanks again!
